### PR TITLE
Add a metrics to monitor grpc job pending queue

### PR DESCRIPTION
**Pull request by jayzhenghan**
### What changes are proposed in this pull request?

Add a metrics to monitor grpc job pending queue

### Why are the changes needed?

You can observe the pressure of the master through this metrics

### CODE REVIEW

**Comment by maobaolong** [comment]
@jayzhenghan Thanks for this useful metrics, before reviewing, would you please fix the style issues

```
/usr/src/alluxio/core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:321: error: Line is longer than 100 characters (found 137).
/usr/src/alluxio/core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:321:103: error: 'typecast' is not followed by whitespace.
```

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java [hunk 1]
```diff
@@ -317,6 +318,9 @@ protected void startServingRPCServer() {
           TimeUnit.MILLISECONDS);
 
       serverBuilder.executor(mRPCExecutor);
+      MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_RPC_PENDING_JOBS.getName(),
```
Comment (maobaolong) [inline core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:321]: As we create mRPCExecutor as a concrete class `ForkJoinPool`, and using the `getQueuedTaskCount` method which only belong to the `ForkJoinPool`, so my suggestion is define mRPCExecutor as the ForkJoinPool.

**Review by maobaolong — CHANGES_REQUESTED** [review]
Left a minor comment inline.

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 1 above)
Comment (jayzhenghan) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:321]: > As we create mRPCExecutor as a concrete class `ForkJoinPool`, and using the `getQueuedTaskCount` method which only belong to the `ForkJoinPool`, so my suggestion is define mRPCExecutor as the ForkJoinPool.

I agree with you

Comment (maobaolong) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:321]: Please remove the unused `import java.util.concurrent.ExecutorService;` also

**Comment by maobaolong** [comment]
@jayzhenghan LGTM, @yuzhu Could you please take another look?

**Review by yuzhu — APPROVED** [review]
LGTM, thanks for the improvement

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 1 above)
Comment (ggezer) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:320]: JFYI; `ForkJoinPool` might go away soon.

Comment (apc999) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java:320]: > JFYI; `ForkJoinPool` might go away soon.

shall we leave a TODO or doc when defining `mRPCExecutor` ?

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 2]
```diff
@@ -368,6 +368,11 @@ public MetricKey build() {
           .setDescription("An estimate of the blocks heap size")
           .setMetricType(MetricType.GAUGE)
           .build();
+  public static final MetricKey MASTER_RPC_PENDING_JOBS =
+      new Builder("Master.RpcPendingJobs")
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:372]: How about `Master.RpcQueueLength`? is it accurate?

Comment (jayzhenghan) [reply core/common/src/main/java/alluxio/metrics/MetricKey.java:372]: RpcCallQueueLength？

Comment (maobaolong) [reply core/common/src/main/java/alluxio/metrics/MetricKey.java:372]: `Master.RpcQueueLength` is better. RpcCallQueueLength is from hdfs, and there is call in the context of hdfs, for Alluxio, just call it `RpcQueueLength`.

Comment (jayzhenghan) [reply core/common/src/main/java/alluxio/metrics/MetricKey.java:372]: ok👌

File: core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java (hunk 1 above)
Comment (jayzhenghan) [reply core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java]: @apc999 How should todo be described

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**